### PR TITLE
Development merge

### PR DESCRIPTION
**Pull request by lezrak**
Basic implementations for crucial classes were made
Endpoints for core flow were created
Test configuration with JPA Mapping test was created
Swagger documentation generation was configured
Secuirty configuration for development purposes was added
Sending gathered SimSteps to AI service added and Scheduled

### CODE REVIEW

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    public Long postSimulation(Long playerId) {
+        Optional<Player> player = playerRepository.findById(playerId);
+        if (player.isEmpty()) throw new RuntimeException("Player not found for id = " + playerId);
+        return simulationRepository.save(new Simulation(player.get())).getId();
+    }
+
+    @Override
+    public void patchSimulation(Long simulationId, SimulationDTO simulationDTO) {
+        if (!simulationRepository.existsById(simulationId))
+            throw new RuntimeException("Simulation not found for id = " + simulationId);
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: EntityNotFoundException?

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        Simulation simulation = simulationRepository.findSimulationById(simulationId);
+        simulation.update(simulationDTO);
+        simulationRepository.save(simulation);
+    }
+
+    @Override
+    public void addModel(Long simulationId, AIModel aiModel) {
+        if (!simulationRepository.existsById(simulationId))
+            throw new RuntimeException("Simulation not found for id = " + simulationId);
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: EntityNotFoundException?

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        Simulation simulation = simulationRepository.findSimulationById(simulationId);
+        simulation.setAiModel(aiModel);
+        simulationRepository.save(simulation);
+    }
+
+    @Override
+    public AIModel getModel(Long simulationId) {
+        if (!simulationRepository.existsById(simulationId))
+            throw new RuntimeException("Simulation not found for id = " + simulationId);
```
Comment (BartoszF) [inline src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: EntityNotFoundException?

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1 +1,7 @@
+spring.datasource.url=jdbc:postgresql://ec2-54-247-72-30.eu-west-1.compute.amazonaws.com:5432/d8d48tlo69oqos?user=jyfqgxzqedprko&password=c965ed4a991d855bb5d7e07de9cbe5131c03b2286d0699740a3660df2262a0bc
```
Comment (BartoszF) [inline src/main/resources/application.properties]: Security breach....
Datasource url should be filled via property

Comment (lezrak) [reply src/main/resources/application.properties]: Done.

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java (hunk 3 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: Not crucial now, will be done in future versions.

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java (hunk 2 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: Not crucial now, will be done in future versions.

File: src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java (hunk 1 above)
Comment (lezrak) [reply src/main/java/pl/neurocars/backend/simulation/SimulationServiceImpl.java]: Not crucial now, will be done in future versions.

**Pull request merged**